hooks/useStakingData.ts

```diff
@@ -207,70 +207,67 @@ const useStakingData = () => {
 		return periods;
 	}, [epochPeriod]);
 
-	data = [];
-
 	const { refetch: resetVesting, data: vestingSchedules } = useContractRead({
 		...rewardEscrowContract,
 		functionName: 'getVestingSchedules',
 		args: [walletAddress ?? undefined, 0, 1000],
 		watch: false,
 		enabled: !!walletAddress,
 		select: (data) => data.filter((d) => d.escrowAmount.gt(0)),
-		onError(error) {
-			if (error) logError(error);
-		},
 	});
 
-	vestingSchedules &&
-		vestingSchedules.length > 0 &&
-		vestingSchedules.forEach((d: VestingEntry) => {
-			data.push({
-				id: Number(d.entryID),
-				date: moment(Number(d.endTime) * 1000).format('MM/DD/YY'),
-				time: formatTruncatedDuration(d.endTime - new Date().getTime() / 1000),
-				vestable: d.endTime * 1000 > Date.now() ? 0 : Number(d.escrowAmount / 1e18),
-				amount: Number(d.escrowAmount / 1e18),
-				fee: d.endTime * 1000 > Date.now() ? Number(d.escrowAmount / 1e18) : 0,
-				status: d.endTime * 1000 > Date.now() ? 'VESTING' : 'VESTED',
+	const escrowRows: EscrowRow[] = useMemo(() => {
+		if (vestingSchedules && vestingSchedules.length > 0) {
+			return vestingSchedules.map((d: VestingEntry) => {
+				return {
+					id: Number(d.entryID),
+					date: moment(Number(d.endTime) * 1000).format('MM/DD/YY'),
+					time: formatTruncatedDuration(d.endTime - new Date().getTime() / 1000),
+					vestable: d.endTime * 1000 > Date.now() ? 0 : Number(d.escrowAmount / 1e18),
+					amount: Number(d.escrowAmount / 1e18),
+					fee: d.endTime * 1000 > Date.now() ? Number(d.escrowAmount / 1e18) : 0,
+					status: d.endTime * 1000 > Date.now() ? 'VESTING' : 'VESTED',
+				};
 			});
-		});
+		}
+		return [];
+	}, [vestingSchedules]);
 
-	const contracts = data.map((d) => {
-		return {
-			...rewardEscrowContract,
-			functionName: 'getVestingEntryClaimable',
-			args: [walletAddress ?? undefined, d.id],
-			enabled: !!walletAddress,
-		};
-	});
+	const vestingEntries = useMemo(() => {
+		return escrowRows.map((d) => {
+			return {
```

Comment:
Not an issue but just fyi often a cleaner solution than multiple returns is you can just do:
```
return escrowRows.map((d) => ({
    ...rewardEscrowContract,
    functionName: 'getVestingEntryClaimable',
    args: [walletAddress ?? undefined, d?.id],
    enabled: !!walletAddress,
}));
```

Reply:
Great to know.